### PR TITLE
Add Fortune extension

### PR DESCRIPTION
**Pull request by netsgnut**
This PR proposes adding a `fortune` extension. When installed and enabled, `fortune` will `print a random, hopefully interesting, adage`. Selecting it copies the line to clipboard.

### CODE REVIEW

File: Fortune.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    item = Item(
+        id=__prettyname__,
+        icon=iconPath,
+        text=__prettyname__,
+        completion=query.rawString
+    )
+    try:
+        item.text = sp.check_output(['fortune', '-s'],
+                                    stderr=sp.STDOUT).decode().strip()
+        item.subtext = "Copy this random, hopefully interesting, adage"
```
Comment (tomsquest) [inline Fortune.py]: `subtext` could be set directly in `Item()`

File: Fortune.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+def generateFortune(query):
+    item = Item(
+        id=__prettyname__,
+        icon=iconPath,
+        text=__prettyname__,
+        completion=query.rawString
+    )
+    try:
+        item.text = sp.check_output(['fortune', '-s'],
+                                    stderr=sp.STDOUT).decode().strip()
```
Comment (tomsquest) [inline Fortune.py]: Why `stderr=sp.STDOUT`?

Comment (netsgnut) [reply Fortune.py]: Indeed it should be not required. Was fiddling around the error handling but left this line there.

File: Fortune.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    if query.isTriggered:
+        newFortune = generateFortune()
+        if newFortune is not None:
+            return getFortuneItem(query, newFortune)
+
+
+def generateFortune():
+    try:
+        fortuneOutput = sp.check_output(["fortune", "-s"]).decode().strip()
+        return fortuneOutput
```
Comment (tomsquest) [inline Fortune.py]: `fortuneOutput` can be inlined

File: Fortune.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+__iid__ = "PythonInterface/v0.1"
+__prettyname__ = "Fortune"
+__version__ = "1.0"
+__trigger__ = "fortune"
+__author__ = "Kelvin Wong"
+__dependencies__ = ["fortune"]
+
+if which("fortune") is None:
+    raise("'fortune' is not in $PATH.")
```
Comment (tomsquest) [inline Fortune.py]: Tried the extension **without** fortune being installed, got this error:

```bash
12:10:56 [WARN] [Fortune.py] TypeError: exceptions must derive from BaseException

At:
  ./Fortune.py(17): <module>
  <frozen importlib._bootstrap>(219): _call_with_frames_removed
  <frozen importlib._bootstrap_external>(678): exec_module
```

So I think `raise` is not enough here for proper error handling.

File: Fortune.py (hunk 3 above)
Comment (netsgnut) [reply Fortune.py]: True. Updated with a9f52a6. Thanks.

**Comment by ManuelSchneid3r** [comment]
Cool. :+1:

**Comment by ManuelSchneid3r** [comment]
I am somehow thinking about writing a generic CLI wrapper that allows to specify a trigger and a commandline. A lot of python extensions are basically CLI wrappers. So this would make things easier I guess.

**Comment by netsgnut** [comment]
Would love to see that happen. With that it sounds many of the boilerplate code can be deprecated.

**Pull request merged**